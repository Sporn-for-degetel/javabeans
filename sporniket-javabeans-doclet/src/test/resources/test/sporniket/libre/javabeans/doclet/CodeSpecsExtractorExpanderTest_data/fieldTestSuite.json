{
	"fields": [
		{
			"refType": "boolean",
			"valIsPackagePrivate": true,
			"valIsPublic": false,
			"valName": "myBoolean",
			"valQualifiedName": "TheClass.myBoolean"
		},
		{
			"refType": "byte",
			"valIsPackagePrivate": true,
			"valIsPublic": false,
			"valName": "myByte",
			"valQualifiedName": "TheClass.myByte"
		},
		{
			"refType": "char",
			"valIsPackagePrivate": true,
			"valIsPublic": false,
			"valName": "myChar",
			"valQualifiedName": "TheClass.myChar"
		},
		{
			"refType": "int",
			"valIsPackagePrivate": true,
			"valIsPublic": false,
			"valName": "myInt",
			"valQualifiedName": "TheClass.myInt"
		},
		{
			"refType": "long",
			"valIsPackagePrivate": true,
			"valIsPublic": false,
			"valName": "myLong",
			"valQualifiedName": "TheClass.myLong"
		},
		{
			"refType": "float",
			"valIsPackagePrivate": true,
			"valIsPublic": false,
			"valName": "myFloat",
			"valQualifiedName": "TheClass.myFloat"
		},
		{
			"refType": "double",
			"valIsPackagePrivate": true,
			"valIsPublic": false,
			"valName": "myDouble",
			"valQualifiedName": "TheClass.myDouble"
		},
		{
			"refType": "short",
			"valIsPackagePrivate": true,
			"valIsPublic": false,
			"valName": "myShort",
			"valQualifiedName": "TheClass.myShort"
		},
		{
			"refType": "short",
			"valIsPackagePrivate": true,
			"valIsPublic": false,
			"valName": "unprefixed",
			"valQualifiedName": "TheClass.unprefixed"
		},
		{
			"refType": "short",
			"valIsPackagePrivate": true,
			"valIsPublic": false,
			"valName": "my",
			"valQualifiedName": "TheClass.my"
		},
		{
			"refType": "java.lang.String[][]",
			"valIsPackagePrivate": true,
			"valIsPublic": false,
			"valName": "arrayOfStrings",
			"valQualifiedName": "TheClass.arrayOfStrings"
		}
	],
	"classes": [
		{
			"refFields": [
				"TheClass.myBoolean",
				"TheClass.myByte",
				"TheClass.myChar",
				"TheClass.myInt",
				"TheClass.myLong",
				"TheClass.myFloat",
				"TheClass.myDouble",
				"TheClass.myShort",
				"TheClass.unprefixed",
				"TheClass.my",
				"TheClass.arrayOfStrings"
			],
			"refInterfaces": [],
			"refSuperclass": "java.lang.Object",
			"valQualifiedName": "TheClass"
		}
	]
}