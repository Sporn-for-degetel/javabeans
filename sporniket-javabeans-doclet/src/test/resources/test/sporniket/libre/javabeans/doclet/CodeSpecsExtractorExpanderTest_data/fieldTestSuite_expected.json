{
	"fields": [
		{
			"arrayMarker":"",
			"nameForField": "Boolean",
			"directlyRequired": true,
			"annotations": [],
			"booleanGetter": true,
			"fieldPrefix": "my",
			"nameForAccessor": "Boolean",
			"typeInvocation": "boolean"
		},
		{
			"arrayMarker":"",
			"nameForField": "Byte",
			"directlyRequired": true,
			"annotations": [],
			"booleanGetter": false,
			"fieldPrefix": "my",
			"nameForAccessor": "Byte",
			"typeInvocation": "byte"
		},
		{
			"arrayMarker":"",
			"nameForField": "Char",
			"directlyRequired": true,
			"annotations": [],
			"booleanGetter": false,
			"fieldPrefix": "my",
			"nameForAccessor": "Char",
			"typeInvocation": "char"
		},
		{
			"arrayMarker":"",
			"nameForField": "Int",
			"directlyRequired": true,
			"annotations": [],
			"booleanGetter": false,
			"fieldPrefix": "my",
			"nameForAccessor": "Int",
			"typeInvocation": "int"
		},
		{
			"arrayMarker":"",
			"nameForField": "Long",
			"directlyRequired": true,
			"annotations": [],
			"booleanGetter": false,
			"fieldPrefix": "my",
			"nameForAccessor": "Long",
			"typeInvocation": "long"
		},
		{
			"arrayMarker":"",
			"nameForField": "Float",
			"directlyRequired": true,
			"annotations": [],
			"booleanGetter": false,
			"fieldPrefix": "my",
			"nameForAccessor": "Float",
			"typeInvocation": "float"
		},
		{
			"arrayMarker":"",
			"nameForField": "Double",
			"directlyRequired": true,
			"annotations": [],
			"booleanGetter": false,
			"fieldPrefix": "my",
			"nameForAccessor": "Double",
			"typeInvocation": "double"
		},
		{
			"arrayMarker":"",
			"nameForField": "Short",
			"directlyRequired": true,
			"annotations": [],
			"booleanGetter": false,
			"fieldPrefix": "my",
			"nameForAccessor": "Short",
			"typeInvocation": "short"
		},
		{
			"arrayMarker":"",
			"nameForField": "Unprefixed",
			"directlyRequired": true,
			"annotations": [],
			"booleanGetter": false,
			"fieldPrefix": "my",
			"nameForAccessor": "Unprefixed",
			"typeInvocation": "short"
		},
		{
			"arrayMarker":"",
			"nameForField": "My",
			"directlyRequired": true,
			"annotations": [],
			"booleanGetter": false,
			"fieldPrefix": "my",
			"nameForAccessor": "My",
			"typeInvocation": "short"
		},
		{
			"arrayMarker":"[][]",
			"nameForField": "ArrayOfStrings",
			"directlyRequired": true,
			"annotations": [],
			"booleanGetter": false,
			"fieldPrefix": "my",
			"nameForAccessor": "ArrayOfStrings",
			"typeInvocation": "String"
		}
	]
}